### PR TITLE
react State  Feature/home page

### PR DESCRIPTION
**Pull request by Farinoxa**

### CODE REVIEW

File: src/modules/Home/components/Content/index.jsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+function Content() {
+  const [inputValue, setInputValue] = useState('');
+  const [chipsValue, setChipsValue] = useState('');
+
+  const filtredArticles = articles.filter((article) => {
+    return article.title.toLowerCase().includes(inputValue.toLowerCase());
+  });
+  const filtredArticlesByCategory = filtredArticles.filter((article) => {
+    return article.category.toLowerCase().includes(chipsValue.toLowerCase());
+  });
```
Comment (des1nteresado) [inline src/modules/Home/components/Content/index.jsx]: You can use 2 conditions in one filter

File: src/modules/shared/Chips/index.jsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        label="All"
+        onClick={() => handleChipsChange('')}
+      />
+      {chipLabels.map((label) => (
+        <Chip
+          id="chip"
+          key={label}
+          clickable
+          label={label}
+          onClick={() => handleChipsChange(label)}
```
Comment (des1nteresado) [inline src/modules/shared/Chips/index.jsx]: It is good UX practise to show selected category.

File: src/modules/Home/components/Content/index.jsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+import { SearchOutlined } from '@material-ui/icons';
+import Article from '../Articles';
+import articles from '../../constants';
+import Chips from '../../../shared/Chips';
+
+import './Content.css';
+
+function Content() {
+  const [inputValue, setInputValue] = useState('');
+  const [chipsValue, setChipsValue] = useState('');
```
Comment (des1nteresado) [inline src/modules/Home/components/Content/index.jsx]: You can combine these states into the one object
`const [appliedFilters, setAppliedFilters] = useState({search: '', filter: ''});`

File: src/modules/Home/components/Content/index.jsx (hunk 1 above)
Comment (Farinoxa) [reply src/modules/Home/components/Content/index.jsx]: the filter function has been redesigned according to the conditions

File: src/modules/shared/Chips/index.jsx (hunk 2 above)
Comment (Farinoxa) [reply src/modules/shared/Chips/index.jsx]: the chips are highlighted when selected

File: src/modules/Home/components/Content/index.jsx [hunk 4]
```diff
@@ -98,7 +104,7 @@ function Content() {
               className={classes.searchInput}
               variant="outlined"
               placeholder="Search"
-              onChange={(event) => setInputValue(event.target.value)}
+              onChange={(event) => setChipsArr({ search: event.target.value })}
```
Comment (des1nteresado) [inline src/modules/Home/components/Content/index.jsx]: `onChange={(event) => setChipsArr(filterValue => { ...filterValue, search: event.target.value })}`

Comment (des1nteresado) [reply src/modules/Home/components/Content/index.jsx]: You should modify only what you really need, not the whole state.

File: src/modules/Home/components/Content/index.jsx (hunk 3 above)
Comment (Farinoxa) [reply src/modules/Home/components/Content/index.jsx]: combined the states into one object

File: src/modules/Home/components/Content/index.jsx (hunk 4 above)
Comment (Farinoxa) [reply src/modules/Home/components/Content/index.jsx]: applied the spread operator

File: src/modules/Home/components/Content/index.jsx [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+import { SearchOutlined } from '@material-ui/icons';
+import Article from '../Articles';
+import articles from '../../constants';
+import Chips from '../../../shared/Chips';
+import classes from '../../../shared/Chips/Chips.module.css';
+
+import './Content.css';
+
+function Content() {
+  const [chipsArr, setChipsArr] = useState({ search: '', filter: [] });
```
Comment (des1nteresado) [inline src/modules/Home/components/Content/index.jsx]: const [appliedFilter, setAppliedFilter] = useState({ search: '', categories: [] });

File: src/modules/Home/components/Content/index.jsx [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+        article.category.toLowerCase().includes(category.toLowerCase()),
+      ),
+  ];
+
+  const filtredArticles = articles.filter((article) =>
+    filters.every((fn) => fn(article)),
+  );
+
+  const handleChipsChange = (currentCategory) => {
+    setChipsArr((filterValue) => ({ ...filterValue, filter: currentCategory }));
```
Comment (des1nteresado) [inline src/modules/Home/components/Content/index.jsx]: please remove this line

File: src/modules/Home/components/Content/index.jsx (hunk 5 above)
Comment (Farinoxa) [reply src/modules/Home/components/Content/index.jsx]: renamed vars for better readability.

File: src/modules/Home/components/Content/index.jsx (hunk 6 above)
Comment (Farinoxa) [reply src/modules/Home/components/Content/index.jsx]: updater was removed

**Pull request merged**